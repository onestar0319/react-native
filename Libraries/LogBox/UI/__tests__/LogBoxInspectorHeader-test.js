/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const LogBoxInspectorHeader = require('../LogBoxInspectorHeader').default;
const render = require('../../../../jest/renderer');

describe('LogBoxInspectorHeader', () => {
  it('should render no buttons for one total', () => {
    const output = render.shallowRender(
      <LogBoxInspectorHeader
        onSelectIndex={() => {}}
        selectedIndex={0}
        total={1}
        level="warn"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render both buttons for two total', () => {
    const output = render.shallowRender(
      <LogBoxInspectorHeader
        onSelectIndex={() => {}}
        selectedIndex={1}
        total={2}
        level="warn"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render two buttons for three or more total', () => {
    const output = render.shallowRender(
      <LogBoxInspectorHeader
        onSelectIndex={() => {}}
        selectedIndex={0}
        total={1}
        level="warn"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render syntax error header', () => {
    const output = render.shallowRender(
      <LogBoxInspectorHeader
        onSelectIndex={() => {}}
        selectedIndex={0}
        total={1}
        level="syntax"
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
