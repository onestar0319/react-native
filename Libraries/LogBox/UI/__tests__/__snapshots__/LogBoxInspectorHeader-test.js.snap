// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorHeader should render both buttons for two total 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 40,
      },
      Object {
        "backgroundColor": "rgba(250, 186, 48, 1)",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 44,
      }
    }
  >
    <LogBoxInspectorHeaderButton
      disabled={false}
      image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAAaUlEQVRYhe3WsQ2AMAwAwRcs5LEYg3HpYANoQKKgcEEUI/6adM5LbgySfmZsNDeACdiApdEfaQGswH6+Xd1jugc9xYQxxhjz9RhaxwxvDuul3MrAqDyjsozKKnWgXUqdsJcAZgqsTFJ5B7gjUNw0n0HHAAAAAElFTkSuQmCC"
      level="warn"
      onPress={[Function]}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(51, 51, 51, 1)",
            "fontSize": 16,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Log 2 of 2
      </Text>
    </View>
    <LogBoxInspectorHeaderButton
      disabled={false}
      image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAAZElEQVRYhe3WsQmAQAxA0Y8ulLEcw3HtdANtBNvzCJjD/5pUgQ9pApJ+Zu7YCWABDmDLzemzA+c94+MW4AkqExUY1caoVka1GibqlSm7qJJSJzPGGGMylYqBgi9sACtFYiQN7wKC6VDcJ7tlpQAAAABJRU5ErkJggg=="
      level="warn"
      onPress={[Function]}
    />
  </View>
</View>
`;

exports[`LogBoxInspectorHeader should render no buttons for one total 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 40,
      },
      Object {
        "backgroundColor": "rgba(250, 186, 48, 1)",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 44,
      }
    }
  >
    <LogBoxInspectorHeaderButton
      disabled={true}
      image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAAaUlEQVRYhe3WsQ2AMAwAwRcs5LEYg3HpYANoQKKgcEEUI/6adM5LbgySfmZsNDeACdiApdEfaQGswH6+Xd1jugc9xYQxxhjz9RhaxwxvDuul3MrAqDyjsozKKnWgXUqdsJcAZgqsTFJ5B7gjUNw0n0HHAAAAAElFTkSuQmCC"
      level="warn"
      onPress={[Function]}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(51, 51, 51, 1)",
            "fontSize": 16,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Log 1 of 1
      </Text>
    </View>
    <LogBoxInspectorHeaderButton
      disabled={true}
      image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAAZElEQVRYhe3WsQmAQAxA0Y8ulLEcw3HtdANtBNvzCJjD/5pUgQ9pApJ+Zu7YCWABDmDLzemzA+c94+MW4AkqExUY1caoVka1GibqlSm7qJJSJzPGGGMylYqBgi9sACtFYiQN7wKC6VDcJ7tlpQAAAABJRU5ErkJggg=="
      level="warn"
      onPress={[Function]}
    />
  </View>
</View>
`;

exports[`LogBoxInspectorHeader should render syntax error header 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 40,
      },
      Object {
        "backgroundColor": "rgba(243, 83, 105, 1)",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 44,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(51, 51, 51, 1)",
            "fontSize": 16,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Failed to compile
      </Text>
    </View>
  </View>
</View>
`;

exports[`LogBoxInspectorHeader should render two buttons for three or more total 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 40,
      },
      Object {
        "backgroundColor": "rgba(250, 186, 48, 1)",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 44,
      }
    }
  >
    <LogBoxInspectorHeaderButton
      disabled={true}
      image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAAaUlEQVRYhe3WsQ2AMAwAwRcs5LEYg3HpYANoQKKgcEEUI/6adM5LbgySfmZsNDeACdiApdEfaQGswH6+Xd1jugc9xYQxxhjz9RhaxwxvDuul3MrAqDyjsozKKnWgXUqdsJcAZgqsTFJ5B7gjUNw0n0HHAAAAAElFTkSuQmCC"
      level="warn"
      onPress={[Function]}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(51, 51, 51, 1)",
            "fontSize": 16,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Log 1 of 1
      </Text>
    </View>
    <LogBoxInspectorHeaderButton
      disabled={true}
      image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAAZElEQVRYhe3WsQmAQAxA0Y8ulLEcw3HtdANtBNvzCJjD/5pUgQ9pApJ+Zu7YCWABDmDLzemzA+c94+MW4AkqExUY1caoVka1GibqlSm7qJJSJzPGGGMylYqBgi9sACtFYiQN7wKC6VDcJ7tlpQAAAABJRU5ErkJggg=="
      level="warn"
      onPress={[Function]}
    />
  </View>
</View>
`;
